airflow/providers/google/cloud/hooks/cloud_sql.py

```diff
@@ -805,6 +858,84 @@ def __init__(
+                    cert_name=cert_name, cert_path=original_cert_path, cert_value=cert_value
+                )
+            return self._ssl_cert_temp_files[cert_name].name
+        return None
+
+    def _get_cert_from_secret(self, cert_name: str) -> str | None:
+        if self.ssl_secret_id:
+            secret_hook = GoogleCloudSecretManagerHook(
+                gcp_conn_id=self.gcp_conn_id, impersonation_chain=self.impersonation_chain
+            )
+            secret: AccessSecretVersionResponse = secret_hook.access_secret(
+                project_id=self.project_id,
+                secret_id=self.ssl_secret_id,
+            )
+            secret_data = json.loads(base64.b64decode(secret.payload.data))
+            if cert_name in secret_data:
+                return secret_data[cert_name]
+            else:
+                raise AirflowException(
+                    "Invalid secret format. Expected dictionary with keys: `sslcert`, `sslkey`, `sslrootcert`"
+                )
+        return None
+
+    def _set_temporary_ssl_file(
+        self, cert_name: str, cert_path: str | None = None, cert_value: str | None = None
+    ) -> str | None:
+        """Save the certificate as a temporary file.
+
+        This method was implemented in order to overcome psql connection error caused by excessive file
+        permissions: "private key file "..." has group or world access; file must have permissions
+        u=rw (0600) or less if owned by the current user, or permissions u=rw,g=r (0640) or less if owned
+        by root". NamedTemporaryFile enforces using exactly one of create/read/write/append mode so the
+        created file obtains least required permissions "-rw-------" that satisfies the rules.
+
+        :param cert_name: Required. Name of the certificate (one of sslcert, sslkey, sslrootcert).
+        :param cert_path: Optional. Path to the certificate.
+        :param cert_value: Optional. The certificate content.
+
+        :returns: The path to the temporary certificate file.
+        """
+        if all([cert_path, cert_value]):
+            raise AirflowException(
+                "Both parameters were specified: `cert_path`, `cert_value`. Please use only one of them."
+            )
+        if any([cert_path, cert_value]):
+            _temp_file = NamedTemporaryFile(mode="w+b", prefix="/tmp/certs/")
+            if cert_path:
+                with open(cert_path, "rb") as cert_file:
+                    _temp_file.write(cert_file.read())
+            elif cert_value:
+                _temp_file.write(cert_value.encode("ascii"))
+            _temp_file.flush()
+            self._ssl_cert_temp_files[cert_name] = _temp_file
+            self.log.info(
+                "Copied the certificate '%s' into a temporary file '%s'", cert_name, _temp_file.name
+            )
+            return _temp_file.name
+        self.log.info("Neither cert path and cert value provided. Nothing to save.")
```

Comment:
You could move this to the top as the first condition:
```suggestion
        if not any([cert_path, cert_value])
            self.log.info("Neither cert path and cert value provided. Nothing to save.")
            return None
```

Reply:
fixed